pprl/layers/layer_wrappers.py

```diff
@@ -0,0 +1,546 @@
+        self.param_attr = param_attr
+        self.bias_attr = bias_attr
+
+    @property
+    def param_name(self):
+        if self.param_attr:
+            return self.param_attr.name
+        else:
+            return None
+
+    @property
+    def bias_name(self):
+        if self.bias_attr:
+            return self.bias_attr.name
+        else:
+            return None
+
+    def parameters(self):
+        return (self.param_attr, self.bias_attr)
+
+
+def get_set_paras(set_paras):
+    param_name, bias_name = None, None
+    if set_paras is not None:
+        assert (type(set_paras) is tuple) and len(set_paras) == 2
+        param_attr, bias_attr = set_paras
+        if param_attr:
+            param_name = param_attr.name
+        if bias_attr:
+            bias_name = bias_attr.name
+    return param_name, bias_name
+
+
+def check_or_replace_name(name, new_name, attr):
+    name = (new_name if name is None else name)
+    ## if this para is not used
+    if attr == False:
+        return False
+
+    if attr is None:
+        return ParamAttr(name=name)
+
+    assert attr.name is None, \
+        "Do not set parameter name for pprl.layers; leave it as None"
+    attr.name = name
+    return attr
+
+
+def update_attr_name(name, param_name, field_name, attr, suffix,
+                     counter_increase):
+    if name is None:
+        name = field_name
+    else:
+        name = "_" + name
+        field_name = "custom"
+
+    if field_name == "custom":
+        custom_counter = getattr(LayerCounter, field_name)
+        if not name in custom_counter:
+            custom_counter[name] = 0
+        idx = custom_counter[name]
+        if counter_increase:
+            custom_counter[name] += 1
+        setattr(LayerCounter, field_name, custom_counter)
+    else:
+        idx = getattr(LayerCounter, field_name)
+        if counter_increase:
+            setattr(LayerCounter, field_name, idx + 1)
+
+    new_name = "%s_%d%s" % (name, idx, suffix)
+    return check_or_replace_name(param_name, new_name, attr)
+
+
+def create_parameter(shape,
+                     dtype,
+                     attr=None,
+                     is_bias=False,
+                     default_initializer=None,
+                     name=None,
+                     set_paras=None):
+    """
+    Return a function that creates paddle.fluid.layers.create_parameter.
+    """
+
+    param_name, _ = get_set_paras(set_paras)
+    self_name = inspect.stack[0][3]
+    attr = update_attr_name(name, param_name, self_name, attr, ".w", True)
+
+    class CreateParameter_(LayerFunc):
+        def __init__(self):
+            super(CreateParameter_, self).__init__(attr)
+
+        def __call__(self):
+            return layers.create_parameter(
+                shape=shape,
+                dtype=dtype,
+                attr=attr,
+                is_bias=is_bias,
+                default_initializer=default_initializer)
+
+    return CreateParameter_()
```

Comment:
The way to use this create_parameter() is not intuitive. Should find a better way for doing this.